packages/@aws-cdk/aws-codepipeline/lib/cross-region-scaffold-stack.ts

```diff
@@ -0,0 +1,61 @@
+import s3 = require('@aws-cdk/aws-s3');
+import cdk = require('@aws-cdk/cdk');
+import crypto = require('crypto');
+
+/**
+ * Construction properties for {@link CrossRegionScaffoldStack}.
+ */
+export interface CrossRegionScaffoldStackProps {
+  region: string;
+
+  account: string;
+}
+
+/**
+ * A Stack containing resources required for the cross-region CodePipeline functionality to work.
+ */
+export class CrossRegionScaffoldStack extends cdk.Stack {
+  /**
+   * The name of the S3 Bucket used for replicating the Pipeline's artifacts into the region.
+   */
+  public readonly replicationBucket: string;
+
+  constructor(parent?: cdk.App, props?: CrossRegionScaffoldStackProps) {
```

Comment:
Why is parent optional?

Reply:
Because `parent` is optional for `Stack`.